### PR TITLE
Add dev script for summarizing IBU CR

### PR DESCRIPTION
**Pull request by donpenney**

### CODE REVIEW

File: hack/summarize-ibu.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+elif [ "${IDLE}" == "True" ]; then
+    echo "  Idle. Last transition time: $(getTime Idle)"
+elif [ "${IDLE}" == "False" ]; then
+    echo "  Idle False!!!"
+    echo "    Time:    $(getTime Idle)"
+    echo "    Reason:  $(getReason Idle)"
+    echo "    Message: $(getMessage Idle)"
+else
+    echo "  Unhandled state!!!"
+fi
```
Comment (omertuc) [inline hack/summarize-ibu.sh:76]: Can be simplified somewhat

```suggestion
#!/bin/bash
#
# Helper script for monitoring IBU CR during an upgrade
# Parses the IBU CR to display a brief summary of key information
#

json=$(oc get ibu upgrade -o json 2>/dev/null)

if [ -z "${json}" ]; then
    echo "Unable to get IBU CR" >&2
    exit 1
fi

function getCondition {
    local ctype="$1"
    shift
    local field="$1"
    shift
    echo "${json}" | jq -r --arg ctype "${ctype}" --arg field "${field}" '.status.conditions[] | select(.type==$ctype)[$field]'
}

SEED=$(echo "${json}" | jq -r '.spec.seedImageRef.image')
VER=$(echo "${json}" | jq -r '.spec.seedImageRef.version')
STAGE=$(echo "${json}" | jq -r '.spec.stage')

IDLE=$(getCondition Idle status)
PREP_PROG=$(getCondition PrepInProgress status)
PREP_COMPLETE=$(getCondition PrepCompleted status)
UPG_PROG=$(getCondition UpgradeInProgress status)
UPG_COMPLETE=$(getCondition UpgradeCompleted status)

echo "Image:   ${SEED}"
echo "Version: ${VER}"
echo "Stage:   ${STAGE}"
echo "Status:"

if [ "${UPG_COMPLETE}" == "True" ]; then
    echo "  Upgrade completed at: $(getCondition UpgradeCompleted time)"
elif [ "${UPG_COMPLETE}" == "False" ] && [ "${UPG_PROG}" == "False" ]; then
    echo "  Upgrade failed:"
    echo "    Time:    $(getCondition UpgradeCompleted time)"
    echo "    Reason:  $(getCondition UpgradeCompleted reason)"
    echo "    Message: $(getCondition UpgradeCompleted message)"
elif [ "${UPG_PROG}" == "True" ]; then
    echo "  Upgrade in progress at: $(getCondition UpgradeInProgress time)"
elif [ "${PREP_COMPLETE}" == "True" ]; then
    echo "  Prep completed at: $(getCondition PrepCompleted time)"
elif [ "${PREP_COMPLETE}" == "False" ] && [ "${PREP_PROG}" == "False" ]; then
    echo "  Prep failed:"
    echo "    Time:    $(getCondition PrepCompleted time)"
    echo "    Reason:  $(getCondition PrepCompleted reason)"
    echo "    Message: $(getCondition PrepCompleted message)"
elif [ "${PREP_PROG}" == "True" ]; then
    echo "  Prep in progress at: $(getCondition PrepInProgress time)"
elif [ "${IDLE}" == "True" ]; then
    echo "  Idle. Last transition time: $(getCondition Idle time)"
elif [ "${IDLE}" == "False" ]; then
    echo "  Idle False!!!"
    echo "    Time:    $(getCondition Idle time)"
    echo "    Reason:  $(getCondition Idle reason)"
    echo "    Message: $(getCondition Idle message)"
else
    echo "  Unhandled state!!!"
fi
```

**Comment by donpenney** [comment]
We should be able to do something similar with the "brief" display of the CR, ie. show a summarized highest-condition status in "oc get ibu", but maybe this is useful for folks until then

**Review by sakhoury — COMMENTED** [review]
/lgtm

**Review by jc-rh — APPROVED** [review]

**Pull request merged**